ER図・README作成

yuta-nakashima opened the pull request:
https://gyazo.com/9f6340068389dfb3921bccfe64ac5007
ご確認をよろしくお願いいたします。

yuta-nakashima committed:
ER図・README作成

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
-* ...
+usersテーブル
+
+| Column          |Type     |Options  |
```
空だと保存出来ないカラムは「null: false」のオプションを記述しましょう。

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
+| ----------------|---------|-------- |
+| nickname        |string   |         |
+| email           |string   |         |
+| password        |string   |         |
```
```
| password        |string   |         |
```
本実装では、deviseというgemを用いてユーザーを実装していただきます。
deviseではパスワードを保存するカラムは「encrypted_password」とする必要がありますので、修正しましょう。

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
+| last_name_kana  |string   |         |
+| birth_year      |integer  |         |
+| birth_month     |integer  |         |
+| birth_day       |integer  |         |
```
```
| birth_year      |integer  |         |
| birth_month     |integer  |         |
| birth_day       |integer  |         |
```
以下の項目について修正しましょう。

```
・生年月日は一つのカラムにまとめましょう
（3つに別れていると、データ取得時に面倒になるなどの観点）
・データベース には日付を保存する型として、date型が用意されていますので型を変更しましょう。
```

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
+| condition       |text          |null: false                     |
+| delivery_from   |string        |null: false                     |
+| delivery_fee    |string        |null: false                     |
+| delivery_date   |string        |null: false                     |
```
```
| category        |text          |null: false                     |
| condition       |text          |null: false                     |
| delivery_from   |string        |null: false                     |
| delivery_fee    |string        |null: false                     |
| delivery_date   |string        |null: false                     |
```
本実装では、上記カラムはactive_hashというgemを用いて実装いただくかと思います。
active_hashでは、以下の修正が必要になりますのでご確認よろしくお願い致します。
```
・カラム名の末尾に_idを追加
・数字を保存するため、カラムの型をinteger型に変更
```

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
+| city            |string        |null: false                     |
+| address         |string        |null: false                     |
+| building_name   |string        |null: false                     |
+| phone_number    |string        |null: false                     |
```
```
| items           |references    |null: false, foreign_key: true  |
| users           |references    |null: false, foreign_key: true  |
```
上記カラムは購入情報（誰がどの商品を購入したか）を保存するカラムになるかと思います。
配送先情報と購入情報は分けることが可能かと思いますので、別テーブルとして作成しましょう。
※テーブルを分けることで、配送先情報が変更された際に片方のテーブルだけの変更で良いなどのメリットがあります。

また、外部キーのカラム名はモデル単数形になりますので末尾のsを削除しましょう。

matsu987 commented on README.md:
```diff
@@ -1,24 +1,42 @@
+| birth_year      |integer  |         |
+| birth_month     |integer  |         |
+| birth_day       |integer  |         |
+
```
テーブル間のアソシエーションも追加しましょう。
例
```
### Association
（ここに追記していく）
```

matsu987 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

yuta-nakashima committed:
指摘事項修正

usersテーブル
null: falseを追記
password→encrypted passwordに変更
birthdayをdate型に変更しまとめた
アソシエーションの追加
date型を使用して誕生日をまとめた
ActiveHashで使用するカラムに_idを追記
購入情報と配送情報を分けてテーブルに設定

yuta-nakashima committed:
修正

yuta-nakashima commented:
修正しましたので、ご確認をよろしくお願いします。

what
null: falseを追記
password→encrypted passwordに変更
birthdayをdate型に変更しまとめた
アソシエーションの追加
date型を使用して誕生日をまとめた
ActiveHashで使用するカラムに_idを追記
購入情報と配送情報を分けてテーブルに設定

masahirokurauchi commented on README.md:
```diff
@@ -1,24 +1,64 @@
+### Association
+- has_many   :items
+- has_many   :orders
+- has_many   :deliveries
```
今回の設計では「配送先住所は購入管理に紐づいている」というようにして実装を行なっていきたいので、ユーザーと配送先住所のアソシエーションは組まなくてもよいかと考えています。

masahirokurauchi commented on README.md:
```diff
@@ -1,24 +1,64 @@
-* Database initialization
+| Column          |Type          |Options                         |
+| ----------------|--------------|------------------------------- |
+| price           |integer       |null: false                     |
```
価格はItemsテーブルに存在するのでこちらには必要はありません

masahirokurauchi commented on README.md:
```diff
@@ -1,24 +1,64 @@
+### Association
+- belongs_to :item
+- has_many   :deliveries
 
```
ユーザーテーブルとのアソシエーションを組みましょう

masahirokurauchi commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| address         |string        |null: false                     |
+| building_name   |string        |null: false                     |
+| phone_number    |string        |null: false                     |
+| order           |references    |null: false, foreign_key: true  |
+| user            |references    |null: false, foreign_key: true  |
```
こちらは必要ありません。
代わりに購入テーブルの外部キーを設定しましょう

masahirokurauchi reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

yuta-nakashima committed:
再修正依頼

yuta-nakashima commented:
各テーブルのアソシエーション変更を行いました。

また、deliveriesテーブルについて、購入テーブルのアソシエーション・外部キー設定は行っておりましたので、userとのアソシエーションの削除のみ行いました。

ご確認をよろしくお願いします。

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,61 @@
+| Column          |Type          |Options                         |
+| ----------------|--------------|------------------------------- |
+| item            |references    |null: false, foreign_key: true  |
+| delivery        |references    |null: false, foreign_key: true  |
```
こちらuserの外部キー制約カラムにしましょう！
(購入管理はuser、itemのidで行うためです。
deliveriesテーブルにorderの外部キー制約カラムがあるため、ordersテーブルにはdeliveryのカラムが不要になります)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,61 @@
+### Association
+- belongs_to :item
+- belongs_to :user
+- has_one    :deliveries
```
has_oneは単数形にしましょう！
`has_one    :deliveries`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,61 @@
+
+| Column          |Type          |Options                         |
+| ----------------|--------------|------------------------------- |
+| post_code       |integer       |null: false                     |
```
郵便番号はハイフンを必要とすることも考慮してstring型で実装しましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,61 @@
+| Column          |Type          |Options                         |
+| ----------------|--------------|------------------------------- |
+| post_code       |integer       |null: false                     |
+| prefecture      |string        |null: false                     |
```
都道府県のカラムはacrive_hashで実装するため、integer型、_idというカラム名にしましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,61 @@
+| prefecture      |string        |null: false                     |
+| city            |string        |null: false                     |
+| address         |string        |null: false                     |
+| building_name   |string        |null: false                     |
```
今回の実装では、建物番号の入力は任意になるため、not null制約は不要です！

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

yuta-nakashima committed:
再々修正

yuta-nakashima commented:
修正を行いましたので、ご確認をお願いします。

what
ActiveHashの部分を_id integerに変更
post_codeを-が入力されても問題ないようにstringにて設定
has_one部分を単数に変更
orderテーブルはusersとitemsのidを使用するのでdeliveriesテーブルの外部キーを外し、usersテーブルの外部キーを設定

i-date reviewed (approved):
修正ありがとうございます！
修正点を拝見したところ問題ないことを確認できましたので、LGTMとさせていただきます。
以降の実装も引き続きこの調子で進めていきましょう。応援しております！

yuta-nakashima committed:
space調整
